src/main/java/com/yahoo/bullet/storm/QueryCategorizer.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright 2018, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.storm;
+
+import com.yahoo.bullet.querying.Querier;
+import com.yahoo.bullet.record.BulletRecord;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+/**
+ * This categorizes running queries into whether they are done, closed or have exceeded the rate limits. Running queries
+ * are provided as a {@link Map} of String query IDs to non-null, valid, initialized {@link Querier} objects. Use
+ * {@link #categorize(Map, boolean)} for categorizing such queries. The boolean tells the categorizer which method to
+ * use to check if the {@link Querier} is closed. A value of true will use {@link Querier#isClosedForPartition()} and
+ * false will use {@link Querier#isClosed()}. The {@link #categorize(BulletRecord, Map)} method assumes that the data
```

Comment:
This line is hard to understand for me. Line 35-37 explains clearer.

Reply:
Rewrote it.